### PR TITLE
upgrade aws sdk to v1.44.72 to support new region ap-southeast-3

### PR DESCRIPTION
**Pull request by WizardXiao**
### What problem does this PR solve?

Ref https://jira.tidbcloud.com/browse/DM-7377
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
upgrade aws sdk to v1.44.72
```

### CODE REVIEW

File: pkg/apis/go.mod [hunk 1]
```diff
@@ -5,7 +5,7 @@ go 1.16
 require (
 	github.com/BurntSushi/toml v0.3.1
 	github.com/Masterminds/semver v1.4.2
-	github.com/aws/aws-sdk-go v1.30.9 // indirect
+	github.com/aws/aws-sdk-go v1.44.72 // indirect
```
Comment (WangLe1321) [inline pkg/apis/go.mod:8]: Do we need `aws-sdk-go` package in the `apis` package?

Comment (WizardXiao) [reply pkg/apis/go.mod:8]: Maybe called by other package, not our code. i will try to find it.

**Review by grovecai — APPROVED** [review]
LGTM

File: pkg/apis/go.mod (hunk 1 above)
Comment (WizardXiao) [reply pkg/apis/go.mod:8]: $ go mod why -m github.com/aws/aws-sdk-go
# github.com/aws/aws-sdk-go
github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1
github.com/prometheus/prometheus/config
github.com/aws/aws-sdk-go/aws/ec2metadata

**Review by WangLe1321 — APPROVED** [review]

**Pull request merged**